### ISSUE TITLE
nquads parser creates one context per triple in the default graph

### ISSUE THREAD

**Issue by pchampin** [RDFLib/rdflib#535]
In nquads, the 4th component (graph id) of each line can be omitted, meaning that the triple belongs to the default graph.

Currently, the nquads parser puts every such triple in a different context, identified by a fresh BNode, which is clearly wrong.

E.g.:
```python
from rdflib import ConjunctiveGraph, Dataset
from rdflib.parser import StringInputSource

ds = ConjunctiveGraph()

src = StringInputSource("""
<http://example.org/s1> <http://example.org/p1> <http://example.org/o1> .
<http://example.org/s2> <http://example.org/p2> <http://example.org/o2> .
<http://example.org/s3> <http://example.org/p3> <http://example.org/o3> <http://example.org/g3> .
""")

ds.parse(src, format="nquads")

assert len(ds) == 3, len(g)
assert len(list(ds.contexts())) == 2, len(list(ds.contexts()))
```

**Comment by niklasl** [comment]
Closely related to #436. We should coordinate solutions.

**Comment by pchampin** [comment]
You are right, those issues related.
The reason I opened a new one is that, regardless of the chosen solution for handling the default graph,
it is clear that those triples should not end up in *different* contexts.

So those are two different problems, fixing one requires fixing the other. Indeed, we should coordinate.

**Comment by niklasl** [comment]
Yes, I agree, this issue for the nquads parser is distinct.

(I'll also look into coordinating a solution for RDFLib/rdflib-jsonld#34 with this. I wrote down #436 when I originally hit that. But attempting to resolve all parsing and serializing of datasets at once might have been too much.)

**Comment by joernhees** [comment]
hmm, one question wrt. the version in which this will be fixed, e.g. by merging #536 :

as we use http://semver.org, is this a bug fix only (so 4.2.2) or an actual backwards incompatible change (5.x.x)?

**Comment by pchampin** [comment]
The current behaviour of the nquads parser is wrong, to a point that no one
could reasonably rely on it.
So I would go for a bug fix.

**Comment by joernhees** [comment]
fixed in #536
